tests/unit/test_utils.py

```diff
@@ -720,3 +720,40 @@ def test_remove_auth_from_url(auth_url, expected_url):
 def test_redact_password_from_url(auth_url, expected_url):
     url = redact_password_from_url(auth_url)
     assert url == expected_url
+
+
+HAS_LEADING_DIR_TESTS = (
+    ('', True),
+    ('setup.py', False),
+    ('dir/', False),
+    ('pkg-1.0/setup.py', True),
+    ('''
+     setup.py
+     module.py
+     ''', False),
+    ('''
+     pkg-1.0/setup.py
+     pkg-1.0/module.py
+     ''', True),
+    ('''
+     a/setup.py
+     b/module.py
+     ''', False),
+    ('''
+     pkg-1.0
+     pkg-1.0/
+     ''', True),
+    ('''
+     pkg-1.0/
+     pkg-1.0/
+     ''', True),
```

Comment:
Something doesn't seem right about the function implementation since, for example, `["pkg-1.0/", "pkg-1.0/"]` returns True but `["pkg-1.0/"]` returns False (see further up for a test case for the latter).